projects/ngx-maplibre-gl/src/lib/control/navigation-control.directive.ts

```diff
@@ -1,60 +1,64 @@
-import { AfterContentInit, Directive, Host, Input } from '@angular/core';
+import { Directive, afterNextRender, inject, input } from '@angular/core';
 import { NavigationControl } from 'maplibre-gl';
 import { MapService } from '../map/map.service';
 import { ControlComponent } from './control.component';
 
 /**
  * `mglNavigation` - a navigation control directive
- * 
+ *
  * @category Directives
- * 
+ *
  * @see [Navigation](https://maplibre.org/ngx-maplibre-gl/demo/navigation)
  * @see [NavigationControl](https://maplibre.org/maplibre-gl-js/docs/API/classes/NavigationControl)
  */
 @Directive({
   selector: '[mglNavigation]',
   standalone: true,
 })
-export class NavigationControlDirective implements AfterContentInit {
-  /* Init inputs */
-  @Input() showCompass?: boolean;
-  @Input() showZoom?: boolean;
-  @Input() visualizePitch?: boolean;
-
-  constructor(
-    private mapService: MapService,
-    @Host() private controlComponent: ControlComponent<NavigationControl>
-  ) {}
+export class NavigationControlDirective {
+  /* Init injection */
+  private readonly mapService = inject(MapService);
+  private readonly controlComponent = inject<
+    ControlComponent<NavigationControl>
+  >(ControlComponent, { host: true });
 
-  ngAfterContentInit() {
-    this.mapService.mapCreated$.subscribe(() => {
-      if (this.controlComponent.control) {
-        throw new Error('Another control is already set for this control');
-      }
-
-      const options: {
-        showCompass?: boolean;
-        showZoom?: boolean;
-        visualizePitch?: boolean;
-      } = {};
+  /* Init inputs */
+  readonly showCompass = input<boolean>();
+  readonly showZoom = input<boolean>();
+  readonly visualizePitch = input<boolean>();
 
-      if (this.showCompass !== undefined) {
-        options.showCompass = this.showCompass;
-      }
+  constructor() {
+    afterNextRender(() => {
+      this.mapService.mapCreated$.subscribe(() => {
```

Comment:
I see that this logic is duplicated between a lot of directives, can we move it to a single place somehow?

Reply:
The logic is duplicated in some directives but I think moving it to one place makes the code less readable and more complex